File: homework7/AssignmentCode/GymAppCore/ClientApp/src/app/classes/SetRecord.ts
import {Set} from './Set';

export class SetRecord {
    setRecordId: number;
    workoutSessionId: number;
    name: string;
    setId: number;
    exerciseId: number;
    order: number;
    goalRepititions: number;
    completedRepititions: number;
    goalWeight: number;
    completedWeight: number;
    skipped: boolean;

    constructor(set: Set, workoutSessionId: number){
        this.name = set.name;
        this.setId = set.setId;
        this.exerciseId = set.exerciseId
        this.order = set.order;
        this.goalRepititions = set.repititions;
        this.goalWeight = set.weight;
        this.completedRepititions = set.repititions;
        this.completedWeight = set.weight
        this.workoutSessionId = workoutSessionId;
      }
}

File: homework7/AssignmentCode/GymAppData/Models/SetRecord.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace GymAppData.Models
{
    public class SetRecord
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int SetRecordId { get; set; }
        public int SetId { get; set; }
        public int WorkoutSessionId { get; set; }
        [Required]
        [StringLength(250)]
        public string Name { get; set; }
        public int Order { get; set; }
        public int GoalRepititions { get; set; }
        public int CompletedRepititions { get; set; }
        public decimal GoalWeight { get; set; }
        public decimal CompletedWeight { get; set; }
        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime Created { get; set; }
        public DateTime? Updated { get; set; }
        public bool IsDeleted { get; set; }
        public virtual Set Set { get; set; }
        public virtual WorkoutSession WorkoutSession { get; set; }
    }
}
